### PR TITLE
Deployment tooling config file

### PR DESCRIPTION
**Pull request by mjudeikis**
Add `config.yaml` for deployer tooling.
Add Config merge functionality
Add tests to test merge and field parity with templates

### CODE REVIEW

File: test/util/populate/walk.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
+package populate
```
Comment (jim-minter) [inline test/util/populate/walk.go]: I think that the above suggestions should simplify unit test coverage to the point where it's not necessary to add this file, and I would prefer that.

File: pkg/deploy/config.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+// path. Location of the file can be provided using DEPLOy_CONFIG variable.
+func GetConfig(region string) (*RPConfig, error) {
+	configFile := "config.yaml"
+	if os.Getenv("DEPLOY_CONFIG") != "" {
+		configFile = os.Getenv("DEPLOY_CONFIG")
+	}
+	_, err := os.Stat(configFile)
+	if os.IsNotExist(err) {
+		return nil, err
+	}
```
Comment (jim-minter) [inline pkg/deploy/config.go]: not needed - the ReadFile error will tell you

File: pkg/deploy/config.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	MDSDEnvironment              string `yaml:"mdsdEnvironment"`
+}
+
+// GetConfig return RP configuration from config.yaml file in the execution root
+// path. Location of the file can be provided using DEPLOy_CONFIG variable.
+func GetConfig(region string) (*RPConfig, error) {
+	configFile := "config.yaml"
+	if os.Getenv("DEPLOY_CONFIG") != "" {
+		configFile = os.Getenv("DEPLOY_CONFIG")
+	}
```
Comment (jim-minter) [inline pkg/deploy/config.go]: why? can we avoid this?

File: pkg/deploy/deploy.go [hunk 4]
```diff
@@ -307,17 +301,38 @@ func (d *deployer) removeOldScalesets(ctx context.Context) error {
...
-	err = json.Unmarshal(params, parameters)
-	if err != nil {
-		return nil, err
+func (d *deployer) getParameters(template map[string]interface{}) (*arm.Parameters, error) {
+	parameters := arm.Parameters{}
+	parameters.Parameters = map[string]*arm.ParametersParameter{}
+
+	s := reflect.TypeOf(d.config.Configuration)
+	if len(template) > 0 {
+		for paramName, param := range template["parameters"].(map[string]interface{}) {
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply pkg/deploy/deploy.go]: https://github.com/Azure/ARO-RP/blob/18e245b67e983795bf889aeb8902aefdf1c45898/pkg/util/arm/marshal.go#L66 :(

File: pkg/deploy/config.go (hunk 3 above)
Comment (mjudeikis) [reply pkg/deploy/config.go]: I suspect as soon as we hit CI or multiple config files - we will need this.

**Pull request merged**